Options.py

```diff
@@ -950,7 +950,10 @@ def as_dict(self, *option_names: str, casing: str = "snake") -> typing.Dict[str,
                 else:
                     raise ValueError(f"{casing} is invalid casing for as_dict. "
                                      "Valid names are 'snake', 'camel', 'pascal', 'kebab'.")
-                option_results[display_name] = getattr(self, option_name).value
+                value = getattr(self, option_name).value
+                if isinstance(value, set):
+                    value = list(value)
```

Comment:
Sorted is pretty fast and will result in reproducible output.
```suggestion
                    value = sorted(value)
```

Reply:
is it worth it? i benchmarked it and it was slower. i suppose this isn't a code path where it's super vital, i'm just not sure how much it matters since sets are already unsorted by design.

Reply:
Reproducibility is nice for testing (diffs), and sorting is nice for debugging (reading) and generally there may be cases where it's better compressible, either because two instances become identical, or because of common prefixes being close to each other.
Sorted() is fast enough that it doesn't matter.